polaris/hub/client.py

```diff
@@ -150,6 +150,10 @@ def ensure_active_token(self, token: OAuth2Token | None = None) -> bool:
             # As of now, this latest version is not available on Conda though.
             token = self.token
 
+        if token is None:
+            self.token = self.fetch_token()
+            return True
+
         is_active = super().ensure_active_token(token)
         if is_active:
             return True
```

Comment:
**Suggestion**
This preserves the next check, where you need wither a user password set, or an active external auth token, for the fetch to succeed.

```suggestion
        if token:
            is_active = super().ensure_active_token(token)
            if is_active:
                return True
```